serif_swing/src/main/kotlin/serif_swing/App.kt

```diff
@@ -12,6 +12,32 @@ import javax.swing.*
 import javax.swing.filechooser.*;
 import javax.swing.text.*
 import java.io.File
+import javax.sound.sampled.Clip
+import javax.sound.sampled.AudioInputStream
+import javax.sound.sampled.AudioSystem
+
+
+object AudioPlayer {
+    var url = ""
+    val clip = AudioSystem.getClip()
+    fun loadAudio(audio_url: String) {
+        if(url != audio_url) {
+            clip.stop()
+            url = audio_url
+            val inputStream = AudioSystem.getAudioInputStream(File(url).getAbsoluteFile())
+            clip.open(inputStream)
+        }
+    }
+    fun play() {
+        if(clip.isRunning()) {
+            clip.stop()
+        }
+        clip.setFramePosition(0)
+        clip.start()
+    }
+}
+
+var logged_in_user: String = ""
```

Comment:
I think the currently logged in user should be retrieved from MatrixSession, as I think it's returned from login

Reply:
Also agree, just didn't feel like scouring the MatrixSession data for it the night I implemented this stuff.